### DISCUSSION TITLE
How to subscribe to collections

### DISCUSSION THREAD

**Discussion by rfc2822** [bitfireAT/webdav-push#23] [General]
How should subscriptions be registered at the WebDAV server? Our current ideas are

1. like [Managed Attachments](https://www.rfc-editor.org/rfc/rfc8607.html#section-3.3) (POST with `?action=` on collection URL)
2. with a separate URL space for the subscriptions, where they can be managed in a RESTful way
    - REST on `/<collection>/push-subscriptions`
    - REST on `(/<principal>?)/push-subscriptions` (would allow subscribing to multiple collections in one request)
3. (Edit) `POST` on `/<collection>` with subscription info in body, like to [create a WebDAV share](https://datatracker.ietf.org/doc/html/draft-pot-webdav-resource-sharing-04#section-4.3)
4. Other?

**Comment by ArnyminerZ** [comment]
Based on RFC-8030, I'd either provide a custom endpoint such as `/push/subscribe` for example, and then pass the collection ID in the body or some header, for example; or provide an endpoint for each collection such as `/push/collections/<collection>` to which you can POST for example.

**Comment by tcitworld** [comment]
Why not create the subscription directly on the collection itself, and use the payload document type?

```
POST /<collection>
Content-Type: application/xml; charset="utf-8"

<create-subscription xmlns="DAV:Push">
  <transport>
   <!-- ...implementation-details -->
  </transport>
</create-subscription>
```

That's similar to how SabreDAV & Nextcloud create collection shares.

I don't think using the `action` query param is the right way to handle things here. In the case of managed attachments it's only used to avoid describing the action in the payload, which is reserved to the attachment data itself.

**Reply by rfc2822** [reply]
Sounds like an interesting alternative, I'll try to find more information about it!
